### PR TITLE
User bugfixes

### PR DESCRIPTION
**Pull request by stefdworschak**
# Business Analysis 🟠 Code Institute 🟠 Community Project - Pull Request 🎉🎉

## PR contains:
- Making some naming more consistent with the rest of the app
- Using the `flask_login.current_user` model in the views instead of custom `User` method
- Further extending the `User` model to handle register, login and password update functionality

## Breaking changes
- None

### CODE REVIEW

File: app/auth.py [hunk 1]
```diff
@@ -49,16 +41,12 @@ def login():
     form = LoginForm()
     if request.method == "POST":
         # Check if username already exists in db
-        user = mongo.db.users.find_one(
-            {"username": request.form.get("username").lower()})
-  
+        user = user = User.find_one_user(request.form.get("username"))
```
Comment (auxfuse) [inline app/auth.py:44]: Hi @stefdworschak `user = user = User...` is there an extra `user = ` assignment here that doesn't need to be here?

**Review by auxfuse — CHANGES_REQUESTED** [review]
Nice clean up! The change from "company" to "organisation" is definitely a welcome one.

Just one thing to query and open for discussion. 🔎 🤜

File: app/auth.py (hunk 1 above)
Comment (stefdworschak) [reply app/auth.py]: Nope. No idea what happened there :rofl:! Removed it now.

**Review by auxfuse — APPROVED** [review]
Thanks for the changes Stefan! And for adding in the Cross Site Request Forgery Tokens 🤜

LGTM 🎉

**Pull request merged**